datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -860,21 +864,31 @@ def refresh_source_node(
     current_revision = source_node.current
 
     # Get the latest columns for the source node's table from the query service
-    columns = query_service_client.get_columns_for_table(
-        current_revision.catalog.name,
-        current_revision.schema_,  # type: ignore
-        current_revision.table,  # type: ignore
-        current_revision.catalog.engines[0]
-        if len(current_revision.catalog.engines) >= 1
-        else None,
-    )
-
-    # Check if any of the columns have changed (only continue with update if they have)
-    column_changes = {col.identifier() for col in current_revision.columns} != {
-        col.identifier() for col in columns
-    }
-    if not column_changes:
-        return source_node
+    new_columns = []
+    try:
+        new_columns = query_service_client.get_columns_for_table(
+            current_revision.catalog.name,
+            current_revision.schema_,  # type: ignore
+            current_revision.table,  # type: ignore
+            current_revision.catalog.engines[0]
+            if len(current_revision.catalog.engines) >= 1
+            else None,
+        )
+    except DJDoesNotExistException:
+        # continue with the update, if the table was not found
+        pass
+
+    if new_columns:
+        # Check if any of the columns have changed (only continue with update if they have)
+        column_changes = {col.identifier() for col in current_revision.columns} != {
+            col.identifier() for col in new_columns
+        }
+        if not column_changes:
+            return source_node
+    else:
+        # since we don't see any columns, we'll assume the table is gone
+        current_revision.status = NodeStatus.INVALID
+        new_columns = current_revision.columns
```

Comment:
I was debating this yday and I thought that having the status INVALID should be good enough message and having columns still visible could help with troubleshooting in general.

Reply:
Ok makes sense. We can probably have some docs somewhere that tells users that invalid source node = deleted table.

Reply:
@shangyian your other comment re: source nodes are always assumed valid in some places just triggered my memory. We discussed this a while ago that source nodes should always be valid since they serve the base layer of the graph and status (valid/invalid) is purely with respect to "Can DJ successfully build the SQL?". Updating columns on a source node during source node refresh already propagates effects to downstream nodes, I think we should actually maintain this design of source nodes always being valid and have a separate indicator to exist that the external reference to the table cannot be found.

Reply:
@samredai I'll wait with merging this PR, but I have a feeling that if we don't allow source nodes to be invalid somewhere in the code then we should probably adjust it. On the other hand, I also think that because the ultimate test of the DJ sql is during runtime, I think it may not necessarily matter that a query is built on a source node (with an invalid status). The status of the source node may remain for informational purposes, although in such case we should make sure that when a table is deleted/renamed the source node reflects that according to our wishes.